Automattic/jetpack
# CRM: Automations expose attributes to frontend

This PR addresses the need to expose attributes to the frontend.
 As it stands, we only have internal IDs/keys for our attributes (such as operators, values, fields, etc.). This means that to communicate with the front-end, we require a translation layer to bridge our base model's internal architecture and the frontend interface. To address this, this PR introduces a new class named Attribute_Definition, designed to encapsulate the rendering details for each of our steps.

*NOTE:*
- Conditions requiring a field list are annotated with TODO. Currently, we lack a comprehensive list of potential fields for our objects (e.g., contacts, invoices, quotes, and transactions) that can be utilized in automations. Addressing this is beyond the scope of this PR.
- @cleacos recommended renaming `Attribute_Definition` to `Step_Attribute`. During implementation, I found this occasionally confusing since we already possess a property named `$attribute` within our steps. This property (`$attribute`) remains unchanged and continues to store the selected attributes. Meanwhile, `Attribute_Definitions` represent the viable selections available.

## Proposed changes:
* Introduces `Attribute_Definition` class.
* Adds a title to our steps.

### Other information:


## Jetpack product discussion
pbhBOz-3UQ-p2
https://github.com/Automattic/zero-bs-crm/issues/3285


## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
* Review the implementation of the Attribute_Definitions class.
* Ensure that all frontend requirements for rendering conditions are addressed in this PR.
* Thoroughly verify the modifications made in our tests.
